packages/oft-evm/contracts/MintBurnOFTAdapter.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+// External imports
+import { IERC20, IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+// Local imports
+import { IMintableBurnable } from "./interfaces/IMintableBurnable.sol";
+import { OFTCore } from "./OFTCore.sol";
+
+/**
+ * @title MintBurnOFTAdapter
+ * @notice A variant of the standard OFT Adapter that uses an existing ERC20's mint and burn mechanisms for cross-chain transfers.
+ *
+ * @dev Inherits from OFTCore and provides implementations for _debit and _credit functions using a mintable and burnable token.
+ */
+abstract contract MintBurnOFTAdapter is OFTCore {
```

Comment:
Why `OFTCore` and not `OFTAdapter`?  You could then drop the `innerToken` logic as it is already encapsulated in `OFTAdapter`.